### ISSUE TITLE
ci: test_no_block is timing out

### ISSUE THREAD

**Issue by benesch** [MaterializeInc/materialize#5872]
https://buildkite.com/materialize/tests/builds/15033#061df272-c6c0-41e1-8093-ed1fcbfeb3e1

The `test_no_block` test is timing out in CI occasionally. This started pretty recently.

**Comment by benesch** [comment]
@umanwizard any chance you could look into this one, or find someone to?

**Comment by benesch** [comment]
Copy pasting from Slack for posterity. Here's a failed build: https://buildkite.com/materialize/tests/builds/15135#b8c7db90-8e06-4f6c-bd40-ff807fd89b42

Here are the logs:

```
2021-02-26T19:28:03.977372Z  INFO sql: test_no_block: connecting to server
2021-02-26T19:28:03.980695Z  INFO coord::catalog: create table materialize.public.t (u1)
2021-02-26T19:28:03.980910Z  INFO coord::catalog: create index materialize.public.t_primary_idx (u2)
2021-02-26T19:28:03.991189Z  INFO coord::catalog: create table materialize.public.t (u1)
2021-02-26T19:28:03.991467Z  INFO coord::catalog: create index materialize.public.t_primary_idx (u2)
2021-02-26T19:28:03.995500Z  INFO coord::catalog: create table materialize.public.t (u1)
2021-02-26T19:28:03.995709Z  INFO coord::catalog: create index materialize.public.t_primary_idx (u2)
2021-02-26T19:28:03.996977Z  INFO sql: test_no_block: spawning thread
2021-02-26T19:28:03.997295Z  INFO sql: test_no_block: accepting fake schema registry connection
2021-02-26T19:28:04.020475Z  INFO sql: test_no_block: in thread; executing create source
2021-02-26T19:28:04.029364Z ERROR librdkafka: FAIL [thrd:localhost:9092/bootstrap]: localhost:9092/bootstrap: Connect to ipv4#127.0.0.1:9092 failed: Connection refused (after 0ms in state CONNECT)
2021-02-26T19:28:04.040569Z  INFO sql: test_no_block: in thread; create source done
```

And here's my diagnosis:

> so here is my suspicion as to what’s happening. that create source command fails because it is unable to connect to kafka for some spurious reason. therefore materialize never tries to connect to the schema registry, and the test’s listener.accept() call blocks forever.

**Comment by benesch** [comment]
This was fixed in #5954.
